### PR TITLE
Pop a toaster on received message

### PR DESCRIPTION
**Pull request by alimtunc**
Show a toaster on received notification

TODO :
- [x] Don't receive messages when the chat is open (so don't receive your own messages)
- [x] Test with the avatar
- [x] Open the chat when you click on the popup
- [x] Fix bug opening chat trigger toaster
- [x] Filter message allowed to see per user
- [x] Fix can't click on notification side

BONUS : 
- [ ] Reset timeout on hover

https://user-images.githubusercontent.com/18298759/229187462-260c5282-6297-416a-a624-884f61838dba.mov

### CODE REVIEW

File: core/modules/messages/client/messages-popup.hbs.html [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<template name="messagesPopup">
+  <ul class="notifications">
+    {{#each message in messages}}
+      <div id={{message.id}} class="toast">
+        {{> avatarViewer user=message.user}}
+        <div class="right-content">
+          {{#if eq message.type 'global'}}
+            <p class="global-title">Global</p>
```
Comment (ramnes) [inline core/modules/messages/client/messages-popup.hbs.html]: "Global" looks weird and inadequate to me, especially if you send / receive a message in a particular zone. Let's put the zone name instead?

File: core/modules/messages/client/messages-popup.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        // Prevent last notifications to be shown
+        message.createdAt < lastCreated ||
+        Session.get("messagesChannel") === message.channel
+      )
+        return;
+
+      lastCreated = message.createdAt;
+
+      const messages = Session.get("messages");
+      const toatMessage = messages.push({
```
Comment (ramnes) [inline core/modules/messages/client/messages-popup.js]:
```suggestion
      const toastMessage = messages.push({
```

File: core/modules/messages/server/messages.js [hunk 3]
```diff
@@ -96,6 +96,22 @@ Meteor.publish('messages', function (channel) {
   return Messages.find({ channel }, { sort: { createdAt: -1 }, limit });
 });
 
+Meteor.publish('allMessages', () => {
```
Comment (ramnes) [inline core/modules/messages/server/messages.js]: nitpicking: I'd rather rename this record set "messages" and the previous one "channelMessages"

**Review by ramnes — APPROVED** [review]
1. Is there a way to close the popup without opening the chat? If not we probably want to add a cross on the top right of the popup.

2. I'm not sure that we want to add a popup when we're already in the chat, even when the message comes from another channel (or should we remove the toasts of the channel when we click on it?)

LGTM overall

**Review by xsyann — APPROVED** [review]

**Pull request merged**